cmd/gcp-controller-manager/node_annotator.go

```diff
@@ -311,6 +310,44 @@ func extractKubeLabels(instance *compute.Instance) (map[string]string, error) {
 	return parsedLabels, nil
 }
 
+func extractResizeRequestLabel(instance *compute.Instance) *string {
+	const createdByRRKey = "google-compute-mig-resize-request"
+
+	if instance.Metadata == nil {
+		return nil
+	}
+
+	createdByRR, err := findValue(instance.Metadata.Items, createdByRRKey, instance.SelfLink)
+	if err != nil {
+		klog.Warningf("While finding %q key got error: %v", createdByRRKey, err)
+		return nil
+	}
+	if createdByRR == nil {
+		return nil
+	}
+
+	if elements := strings.Split(*createdByRR, "/"); len(elements) > 1 {
+		return &elements[len(elements)-1]
+	}
+
+	klog.Warningf("Retrieved label value %q has unexpected format", *createdByRR)
+	return nil
+}
+
+func findValue(items []*compute.MetadataItems, key, selfLink string) (*string, error) {
```

Comment:
pointer string and error return is kind of unwieldy. Can we return `(value string, found bool)` work here?